mailbox/backup/src/main/java/org/apache/james/mailbox/backup/MailboxWithAnnotations.java

```diff
@@ -0,0 +1,49 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+package org.apache.james.mailbox.backup;
+
+import java.util.List;
+import java.util.Objects;
+
+import org.apache.james.mailbox.model.MailboxAnnotation;
+import org.apache.james.mailbox.store.mail.model.Mailbox;
+
+public class MailboxWithAnnotations {
+    public final Mailbox mailbox;
+    public final List<MailboxAnnotation> annotations;
+
+    public MailboxWithAnnotations(Mailbox mailbox, List<MailboxAnnotation> annotations) {
+        this.mailbox = mailbox;
+        this.annotations = annotations;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
```

Comment:
Usually, we are using this pattern for `equals`
```
    @Override
    public final boolean equals(Object o) {
        if (o instanceof MyClass) {
            MyClass that = (MyClass) o;

            return Objects.equals(this.field1, that.field1)
                && Objects.equals(this.field2 that.field2);
        }
        return false;
    }
```

but there's not a strict consensus on that.